### PR TITLE
Add Github Action workflow to test RIT binaries for `workspace` commands

### PR DESCRIPTION
**Pull request by GuillaumeFalourd**
### Description

Closes: https://github.com/ZupIT/ritchie-cli/issues/957

- Funcional tests using **rit** binaries for each OS, for the `workspace` core commands. Those kind of tests could improve the project quality and future pipeline performances by testing only what matters on each contribution, instead of testing everything every time.

- This workflow will trigger every time one of the paths informed files is updated, by a `PULL REQUEST` event or a `PUSH` event.

- The workflow executes in **parallel** jobs on `windows`, `ubuntu` and `macOS` where each job will:
    - use the make command to generate rit binary for the OS.
    - execute synchronously the following steps: `rit add workspace`, `rit list workspace`, `rit delete workspace` e `rit update workspace`.
    - after each step, a check run is performed using `diff` command to assert the output is as expected.

- Expected outputs need to be added to the `/testdata/gha_workflows` directory, currently on the `.txt` format

### Add-On

- Testing workspace commands, I observed it was possible to update repo commands test as well following the same logic.

#### Workspace

- You can check the workflow execution on [this fork](https://github.com/GuillaumeFalourd/ritchie-cli/blob/master/.github/workflows/test-workspace-commands.yml)

Here are some samples:

Screen Shot 2021-06-28 at 16 31 19

Screen Shot 2021-06-28 at 16 31 26

[Reference](https://github.com/GuillaumeFalourd/ritchie-cli/actions/runs/980144058)

#### Repositories

- You can check the workflow execution on [this fork](https://github.com/GuillaumeFalourd/ritchie-cli/blob/master/.github/workflows/test-repo-commands.yml)

Here are some samples:

Screen Shot 2021-06-28 at 16 44 48

Screen Shot 2021-06-28 at 16 44 53

[Reference](https://github.com/GuillaumeFalourd/ritchie-cli/actions/runs/980217161)

### Changelog

- Adding Github Action workflow performing funcional tests using **rit** binaries for each OS, for the `workspace` core commands.

---------------------------------------------

### This pull request generated the following artifacts.
To test the health and quality of this implementation, download the respective binary for your operating system, unzip and directly run the binary like the examples below.
- **Windows**
 Download the file: **[rit-windows.zip](https://github.com/ZupIT/ritchie-cli/suites/3113674355/artifacts/71275534)**
 Unzip to some folder like: `C:\home\user\downloads\pr972`
 Access the folder: `cd C:\home\user\downloads\pr972`
 Directly call the binary: `.\rit.exe --version` or `.\rit.exe name of formula`
 

 - **Linux**
 Download the file: **[rit-linux.zip](https://github.com/ZupIT/ritchie-cli/suites/3113674355/artifacts/71275532)**
 Unzip to some folder like: `/home/user/downloads/pr972`
 Access the folder: `cd /home/user/downloads/pr972`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`
 

 - **MacOS**
 Download the file: **[rit-macos.zip](https://github.com/ZupIT/ritchie-cli/suites/3113674355/artifacts/71275533)**
 Unzip to some folder like: `/home/user/downloads/pr972`
 Access the folder: `cd /home/user/downloads/pr972`
 Assign execute permission to binary: `chmod +x ./rit`
 Directly call the binary: `./rit --version` or `./rit name of formula`

> Generated at Tue Jun 29 2021 13:33:45 GMT+0000 (Coordinated Universal Time)

### CODE REVIEW

File: .github/workflows/test-repo-commands.yml [hunk 1]
```diff
@@ -115,32 +146,61 @@ jobs:
...
+            command_line: rit delete repo --name="demo"
+            contains: Repository \"demo\" was deleted with success
+            expected_result: PASSED        
+        - name: RIT LIST REPO command (10)
           uses: GuillaumeFalourd/test-cli-commands-action@v1
           with:
             command_line: rit list repo
             assert_file_path: testdata/gha_workflows/repo_workflow/assert1.txt
-            expected_result: PASSED  
+            expected_result: PASSED
```
Comment (lucasdittrichzup) [inline .github/workflows/test-repo-commands.yml:203]: is this space correct?

**Review by lucasdittrichzup — DISMISSED** [review]

File: .github/workflows/test-repo-commands.yml (hunk 1 above)
Comment (GuillaumeFalourd) [reply .github/workflows/test-repo-commands.yml:203]: I checked here and there are different lines with those spaces on the file.
It won't affect the workflow execution, so let me know if you think I should remove it or not.

Comment (lucasdittrichzup) [reply .github/workflows/test-repo-commands.yml:203]: if it's not something that takes time to be done, I believe it's worth sanitizing the file

Comment (GuillaumeFalourd) [reply .github/workflows/test-repo-commands.yml]: Done :)

**Review by lucasdittrichzup — APPROVED** [review]

**Pull request merged**